File: docs/examples/sdks/python/trust.py
from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey
from cryptography.hazmat.primitives import serialization
from hushspec import FileProvider, HushGuard, EvaluationAction
from hushspec.signing import Keyring, sign_policy, verify_policy


def check_trust(policy_path, resolved):
    # Ephemeral test key only. Never use example-generated keys as shared roots.
    key = Ed25519PrivateKey.generate()
    private = key.private_bytes(serialization.Encoding.PEM, serialization.PrivateFormat.PKCS8, serialization.NoEncryption())
    public = key.public_key().public_bytes(serialization.Encoding.PEM, serialization.PublicFormat.SubjectPublicKeyInfo).decode()
    keyring = Keyring.from_public_key(public)
    envelope = sign_policy(resolved, private)
    assert verify_policy(resolved, envelope, keyring=keyring).valid
    other = Ed25519PrivateKey.generate().public_key().public_bytes(serialization.Encoding.PEM, serialization.PublicFormat.SubjectPublicKeyInfo).decode()
    assert not verify_policy(resolved, envelope, keyring=Keyring.from_public_key(other)).valid
    with HushGuard.from_provider(FileProvider(policy_path)) as guard:
        assert not guard.check(EvaluationAction(type="tool_call", target="deploy"))

File: docs/examples/sdks/typescript/trust.mjs
import assert from 'node:assert/strict';
import { FileProvider, HushGuard, generateKeypair, keyringFromPublicKey,
  signPolicy, verifyPolicy } from '@hushspec/core';

export async function checkTrust(policyPath, resolved) {
  // Ephemeral demonstration key. Production trust roots are operator inputs.
  const key = generateKeypair();
  const keyring = keyringFromPublicKey(key.publicKeyPem);
  const envelope = signPolicy(resolved, key.privateKeyPem);
  assert.equal(verifyPolicy(resolved, envelope, {keyring}).ok, true);
  const wrong = keyringFromPublicKey(generateKeypair().publicKeyPem);
  assert.equal(verifyPolicy(resolved, envelope, {keyring:wrong}).ok, false);
  const provider = new FileProvider(policyPath);
  try {
    const guard = await HushGuard.fromProvider(provider);
    assert.equal(guard.check({type:'tool_call', target:'deploy'}), false);
  } finally {
    provider.stop();
  }
}
